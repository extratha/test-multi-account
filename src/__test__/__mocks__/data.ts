export const data = {
  test: 'test'
}