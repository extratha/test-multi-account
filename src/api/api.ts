import { ExampleDataResult, InterpretResult, SubmitLabInterpretsResponse } from "@/types/aiInterpret";
import { SubmitLabInterpretsRequest } from "@/types/interpretInputDataConfig";
import axiosInstance from "@/utils/axios";

export const apiAxios = axiosInstance;

export const getLabExampleList = () => {
  return apiAxios.get<ExampleDataResult[]>("/lab/examples");
};

export const getLabExampleId = (id: string) => {
  return apiAxios.get<InterpretResult>(`/lab/examples/${id}`);
};

export const submitLabInterprets = (data: SubmitLabInterpretsRequest) => {
  return apiAxios.post<SubmitLabInterpretsResponse>("/lab/interprets", data);
};

export const getLabInterpretsByTransactionId = (transactionId: string) => {
  return apiAxios.get<InterpretResult>(`/lab/interprets/${transactionId}`);
};
