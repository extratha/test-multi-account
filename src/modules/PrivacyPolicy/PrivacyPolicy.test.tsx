import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import MockAdapter from "axios-mock-adapter";

import { mockPrivacyPolicyData } from "@/__mocks__/data";
import * as Api from "@/api/api";
import { webPaths } from "@/constant/webPaths";
import { spyUseRouter, SpyUseRouter } from "@/testUtils/navigation";
import { API, flushPromise, render } from "@/testUtils/testUtils";
import axiosInstance from "@/utils/axios";
import PrivacyPolicyModule from ".";

describe("PrivacyPolicyModule", () => {
  let mockApiAdapter: MockAdapter;
  let spyRouter: SpyUseRouter;

  beforeEach(() => {
    spyRouter = spyUseRouter();

    jest.spyOn(Api, "getPrivacyPolicy");
    jest.spyOn(Api, "submitConsent");

    mockApiAdapter = new MockAdapter(axiosInstance);
    mockApiAdapter.onGet(API.PRIVACY_POLICY).reply(200, mockPrivacyPolicyData);
    mockApiAdapter.onPost(API.CONSENT).reply(201);
  });

  const renderPrivacyPolicy = async () => {
    const view = render(<PrivacyPolicyModule />);
    await flushPromise();
    return view;
  };

  it("should render correctly", async () => {
    const { asFragment } = await renderPrivacyPolicy();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should submit consent and navigate to Home page when submit privacy policy agreement success", async () => {
    await renderPrivacyPolicy();

    await userEvent.click(screen.getByTestId("agreement-checkbox"));

    expect(screen.getByTestId("submit-button")).not.toBeDisabled();

    await userEvent.click(screen.getByTestId("submit-button"));

    expect(Api.submitConsent).toHaveBeenCalledTimes(1);
    expect(spyRouter.replace).toHaveBeenCalledWith(webPaths.home);
  });

  it("should redirect to Home page when consent is submitted", async () => {
    mockApiAdapter.onGet(API.PRIVACY_POLICY).reply(200, { ...mockPrivacyPolicyData, isConsent: true });

    await renderPrivacyPolicy();

    expect(spyRouter.replace).toHaveBeenCalledWith(webPaths.home);
  });

  it("should not display consent when fetch consent error", async () => {
    mockApiAdapter.onGet(API.PRIVACY_POLICY).networkError();

    await renderPrivacyPolicy();

    expect(screen.queryByTestId("privacy-policy-consent")).not.toBeInTheDocument();
  });

  it("should not navigate to Home page when submit consent error", async () => {
    mockApiAdapter.onPost(API.CONSENT).networkError();

    await renderPrivacyPolicy();

    await userEvent.click(screen.getByTestId("agreement-checkbox"));
    await userEvent.click(screen.getByTestId("submit-button"));

    expect(spyRouter.push).not.toHaveBeenCalled();
  });
});
