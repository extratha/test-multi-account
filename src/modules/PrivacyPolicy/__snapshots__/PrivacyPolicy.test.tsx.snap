// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivacyPolicyModule should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
  color: unset;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  box-sizing: border-box;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: fixed;
  z-index: 1100;
  top: 0;
  left: auto;
  right: 0;
  background-color: #2196F3;
  color: #fff;
  color: #212121;
  background-color: #FFFFFF;
  box-shadow: 0px 4px 6px -1px rgba(0, 0, 0, 0.1);
}

@media print {
  .emotion-0 {
    position: absolute;
  }
}

.emotion-0>.MuiToolbar-root {
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
  min-height: 48px;
}

@media (min-width:600px) {
  .emotion-1 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

.emotion-2 {
  margin: 0;
  font-size: 16px;
  font-weight: 700;
  line-height: 1.5;
  text-align: center;
}

<header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed emotion-0"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense emotion-1"
    >
      <p
        class="MuiTypography-root MuiTypography-bodyBold emotion-2"
      >
        เลื่อนขึ้นเพื่ออ่านข้อความทั้งหมดและดำเนินการต่อ
      </p>
    </div>
  </header>
  .emotion-0 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
  max-width: 1024px;
  margin: 72px auto 31px;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

.emotion-1 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  color: unset;
  padding: 36px 40px;
  border-radius: 16px;
  box-shadow: none;
}

.emotion-2 {
  margin: 0;
  font-size: 20px;
  font-weight: 700;
  line-height: 28px;
}

.emotion-3 {
  margin: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.12);
  border-bottom-width: thin;
  margin: 8px 0px 24px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-4>:not(style):not(style) {
  margin: 0;
}

.emotion-4>:not(style)~:not(style) {
  margin-top: 24px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
  border-radius: 12px;
  background-color: #FAFAFA;
}

.emotion-6 {
  font-size: 16px;
  line-height: 28px;
}

.emotion-6>h3:first-of-type,
.emotion-6>p:first-of-type {
  margin-top: 0px;
}

.emotion-6>h3:last-of-type,
.emotion-6>p:last-of-type {
  margin-bottom: 0px;
}

.emotion-6 ul {
  padding-left: 24px;
  list-style-type: disc;
}

.emotion-6 li {
  margin: 16px 0px;
}

.emotion-6 ol {
  list-style-type: none;
  counter-reset: item;
  padding: 0px;
}

.emotion-6 ol>li {
  display: table-row;
  counter-increment: item;
  border-collapse: separate;
  border-spacing: 0px 6px;
}

.emotion-6 ol>li:before {
  display: table-cell;
  content: counters(item, ".") ". ";
  padding-right: 4px;
  white-space: nowrap;
}

.emotion-6 ol>ol,
.emotion-6 ol>ul {
  padding-left: 24px;
}

.emotion-6 ol>ol>li>ol,
.emotion-6 ol>ul>li>ol,
.emotion-6 ol>ol>li>ul,
.emotion-6 ol>ul>li>ul {
  padding: 0px;
}

.emotion-6 table {
  width: 100%;
  border-collapse: collapse;
  table-layout: fixed;
  margin: 16px 0px;
}

.emotion-6 table li {
  margin: 0px;
}

.emotion-6 th {
  height: 48px;
  text-align: center;
  color: #ffffff;
  border: 1px solid #000000;
  border-right-color: #ffffff;
  background-color: #002060;
}

.emotion-6 th:last-child {
  border-color: #000000;
}

.emotion-6 td {
  border: 1px solid #000000;
  padding: 8px;
  vertical-align: top;
}

.emotion-8 {
  margin: 0;
  font-size: 18px;
  font-weight: 700;
  line-height: 24px;
  margin-bottom: 24px;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #CFD8DC;
  background-color: #fff;
}

.emotion-9:first-of-type {
  border-radius: 12px 12px 0 0;
}

.emotion-9:last-of-type {
  border-radius: 0 0 12px 12px;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 24px;
}

.emotion-12 {
  max-width: 52px;
  max-height: 52px;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-left: 12px;
}

.emotion-13>:not(style):not(style) {
  margin: 0;
}

.emotion-13>:not(style)~:not(style) {
  margin-top: 4px;
}

.emotion-14 {
  margin: 0;
  font-size: 16px;
  font-weight: 500;
  line-height: 1.5;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-16 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-16::-moz-focus-inner {
  border-style: none;
}

.emotion-16.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-16 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-16:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-16:hover {
    background-color: transparent;
  }
}

.emotion-16.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-17 {
  width: 24px;
  height: 24px;
}

.emotion-18 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-19 {
  height: 0;
  overflow: hidden;
  -webkit-transition: height 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: height 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  visibility: hidden;
}

.emotion-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.emotion-21 {
  width: 100%;
}

.emotion-22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 16px;
}

.emotion-69 {
  margin-top: 24px;
}

.emotion-70 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  margin-left: -11px;
  margin-right: 16px;
}

.emotion-70.Mui-disabled {
  cursor: default;
}

.emotion-70 .MuiFormControlLabel-label.Mui-disabled {
  color: #B0BEC5;
}

.emotion-70>.MuiCheckbox-root {
  color: #cfd8dc;
}

.emotion-70>.Mui-checked {
  color: #2196F3;
}

.emotion-70>.MuiTypography-root {
  font-weight: 500;
  color: #78909c;
}

.emotion-71 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  padding: 9px;
  border-radius: 50%;
  color: rgba(0, 0, 0, 0.6);
}

.emotion-71::-moz-focus-inner {
  border-style: none;
}

.emotion-71.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-71 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-71:hover {
  background-color: rgba(33, 150, 243, 0.04);
}

@media (hover: none) {
  .emotion-71:hover {
    background-color: transparent;
  }
}

.emotion-71.Mui-checked,
.emotion-71.MuiCheckbox-indeterminate {
  color: #2196F3;
}

.emotion-71.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-72 {
  cursor: inherit;
  position: absolute;
  opacity: 0;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.emotion-73 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-75 {
  margin: 0;
  font-family: fontFamily;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
}

.emotion-76 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 16px 0px;
}

.emotion-77 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: fontFamily;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #2196F3;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  font-size: 12px;
  font-weight: 700;
  line-height: 1.5;
  background-color: #2196F3;
  height: 44px;
  width: 100%;
  max-width: 400px;
  border-radius: 8px;
  font-size: 14px;
}

.emotion-77::-moz-focus-inner {
  border-style: none;
}

.emotion-77.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-77 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-77:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(23, 105, 170);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-77:hover {
    background-color: #2196F3;
  }
}

.emotion-77:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-77.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-77.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-77.Mui-disabled {
  background-color: #FAFAFA;
  colors: #B0BEC5;
}

<div
    class="MuiContainer-root MuiContainer-maxWidthLg emotion-0"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 emotion-1"
      data-testid="terms-and-conditions-consent"
    >
      <p
        class="MuiTypography-root MuiTypography-titleLargeBold emotion-2"
      >
        ข้อกำหนดและเงื่อนไข
      </p>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth emotion-3"
      />
      <div
        class="MuiStack-root emotion-4"
      >
        <div
          class="MuiStack-root emotion-5"
        >
          <div
            class="emotion-6"
            data-testid="term-and-conditions-content"
          >
            <h2>
              <strong>
                Playground
              </strong>
            </h2>
          </div>
        </div>
        <div
          class="MuiStack-root emotion-5"
          data-testid="term-and-conditions-service"
        >
          <p
            class="MuiTypography-root MuiTypography-titleBold emotion-8"
          >
            All Services
          </p>
          <div
            class="MuiStack-root emotion-9"
          >
            <div
              class="MuiStack-root emotion-10"
              data-testid="term-and-conditions-service-0"
            >
              <div
                class="MuiBox-root emotion-11"
              >
                <img
                  alt=""
                  class="emotion-12"
                  data-testid="term-and-conditions-service-0-logo"
                  src="https://cdn-dev-playground.cariva.co.th/privacy_policies/images/policy.png"
                />
              </div>
              <div
                class="MuiStack-root emotion-13"
              >
                <p
                  class="MuiTypography-root MuiTypography-bodyMedium emotion-14"
                  data-testid="term-and-conditions-service-0-title"
                >
                  AI Interpret
                </p>
                <svg
                  data-testid="term-and-conditions-service-0-detail-icon"
                />
              </div>
              <div
                class="MuiStack-root emotion-15"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-16"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    class="emotion-17"
                  />
                  <span
                    class="MuiTouchRipple-root emotion-18"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-19"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical emotion-20"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-21"
                >
                  <div
                    class="MuiStack-root emotion-22"
                    data-testid="term-and-conditions-service-0-content"
                  >
                    <div
                      class="emotion-6"
                      data-testid="term-and-conditions-content"
                    >
                      **Service Overview**

AI Interpret provides advanced language translation capabilities...

**Features**

- Real-time translation
- Multi-language support
- High accuracy rates

**Usage Instructions**

1. Select your input and output languages.
2. Speak or type your text.
3. Receive the translated output.
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiStack-root emotion-9"
          >
            <div
              class="MuiStack-root emotion-10"
              data-testid="term-and-conditions-service-1"
            >
              <div
                class="MuiBox-root emotion-11"
              >
                <img
                  alt=""
                  class="emotion-12"
                  data-testid="term-and-conditions-service-1-logo"
                  src="https://cdn-dev-playground.cariva.co.th/privacy_policies/images/policy.png"
                />
              </div>
              <div
                class="MuiStack-root emotion-13"
              >
                <p
                  class="MuiTypography-root MuiTypography-bodyMedium emotion-14"
                  data-testid="term-and-conditions-service-1-title"
                >
                  Health Visualization
                </p>
                <svg
                  data-testid="term-and-conditions-service-1-detail-icon"
                />
              </div>
              <div
                class="MuiStack-root emotion-15"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-16"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    class="emotion-17"
                  />
                  <span
                    class="MuiTouchRipple-root emotion-18"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-19"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical emotion-20"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-21"
                >
                  <div
                    class="MuiStack-root emotion-22"
                    data-testid="term-and-conditions-service-1-content"
                  >
                    <div
                      class="emotion-6"
                      data-testid="term-and-conditions-content"
                    >
                      **Service Overview**

Health Visualization helps you understand your health data through visual representations...

**Features**

- Detailed health metrics
- Trend analysis
- Personalized insights

**Usage Instructions**

1. Upload your health data.
2. Choose the metrics you want to visualize.
3. View the visualized data and insights.
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiStack-root emotion-9"
          >
            <div
              class="MuiStack-root emotion-10"
              data-testid="term-and-conditions-service-2"
            >
              <div
                class="MuiBox-root emotion-11"
              >
                <img
                  alt=""
                  class="emotion-12"
                  data-testid="term-and-conditions-service-2-logo"
                  src="https://cdn-dev-playground.cariva.co.th/privacy_policies/images/policy.png"
                />
              </div>
              <div
                class="MuiStack-root emotion-13"
              >
                <p
                  class="MuiTypography-root MuiTypography-bodyMedium emotion-14"
                  data-testid="term-and-conditions-service-2-title"
                >
                  ASR
                </p>
                <svg
                  data-testid="term-and-conditions-service-2-detail-icon"
                />
              </div>
              <div
                class="MuiStack-root emotion-15"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-16"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    class="emotion-17"
                  />
                  <span
                    class="MuiTouchRipple-root emotion-18"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-19"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical emotion-20"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-21"
                >
                  <div
                    class="MuiStack-root emotion-22"
                    data-testid="term-and-conditions-service-2-content"
                  >
                    <div
                      class="emotion-6"
                      data-testid="term-and-conditions-content"
                    >
                      **Service Overview**

ASR (Automatic Speech Recognition) converts spoken language into text...

**Features**

- High accuracy speech-to-text
- Supports multiple languages
- Real-time processing

**Usage Instructions**

1. Select your language.
2. Speak into the microphone.
3. View the transcribed text.
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiStack-root emotion-9"
          >
            <div
              class="MuiStack-root emotion-10"
              data-testid="term-and-conditions-service-3"
            >
              <div
                class="MuiBox-root emotion-11"
              >
                <img
                  alt=""
                  class="emotion-12"
                  data-testid="term-and-conditions-service-3-logo"
                  src="https://cdn-dev-playground.cariva.co.th/privacy_policies/images/policy.png"
                />
              </div>
              <div
                class="MuiStack-root emotion-13"
              >
                <p
                  class="MuiTypography-root MuiTypography-bodyMedium emotion-14"
                  data-testid="term-and-conditions-service-3-title"
                >
                  Symptom Checker
                </p>
                <svg
                  data-testid="term-and-conditions-service-3-detail-icon"
                />
              </div>
              <div
                class="MuiStack-root emotion-15"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-16"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    class="emotion-17"
                  />
                  <span
                    class="MuiTouchRipple-root emotion-18"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-19"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical emotion-20"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-21"
                >
                  <div
                    class="MuiStack-root emotion-22"
                    data-testid="term-and-conditions-service-3-content"
                  >
                    <div
                      class="emotion-6"
                      data-testid="term-and-conditions-content"
                    >
                      **Service Overview**

Symptom Checker helps you understand potential causes for your symptoms...

**Features**

- Symptom input
- Potential diagnoses
- Suggested next steps

**Usage Instructions**

1. Enter your symptoms.
2. View the potential diagnoses.
3. Follow the suggested next steps for further action.
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <form
        class="emotion-69"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd emotion-70"
          data-testid="agreement-checkbox"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium emotion-71"
            data-testid="agreement-checkbox-item-button"
          >
            <input
              class="PrivateSwitchBase-input emotion-72"
              data-indeterminate="false"
              data-testid="agreement-checkbox-value"
              name="agreement"
              type="checkbox"
              value="false"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-73"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root emotion-18"
            />
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-75"
            data-testid="agreement-checkbox-label"
          >
            ฉันยอมรับเงื่อนไขและข้อตกลงการให้บริการ
          </p>
        </label>
        <div
          class="MuiStack-root emotion-76"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary emotion-77"
            data-testid="submit-button"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            ถัดไป
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
